homeassistant/config.py

```diff
@@ -669,6 +680,116 @@ def async_process_component_config(hass, config, domain):
     return None
 
 
+def check_ha_config_file(config_dir):
+    """Check if Home Assistant configuration file is valid."""
+    from unittest.mock import patch
```

Comment:
We should never use the unittest module in normal code. It is a big red flag of poor design.

Reply:
Suggested another aproach at the patch...

Reply:
Yes, I saw it later. Like it!